anomdetect/src/test/java/com/expedia/adaptivealerting/anomdetect/detect/outlier/algo/forecasting/ForecastingDetectorTest.java

```diff
@@ -94,6 +95,7 @@ public void testDetect_nullForecast() {
 
         val result = (OutlierDetectorResult) detectorUnderTest.detect(metricData);
         assertNotNull(result);
+        assertFalse(result.isWarmup());
```

Comment:
Can we also add a test case for `assertTrue(result.isWarmup())`?

Something like:

```
    @Test
    public void testDetect_warmingUpForecast() {
        initDependencies(true);
        val metricDef = TestObjectMother.metricDefinition();
        val metricData = new MetricData(metricDef, 100.0, Instant.now().getEpochSecond());
        val result = (OutlierDetectorResult) detectorUnderTest.detect(metricData);
        assertTrue(result.isWarmup());
        assertNotNull(result);
    }
```

Above code assumes parameterising the `warmup` setting in `initDependencies()` and calling `initDependencies()` from every test case instead of in `setUp()`.